### PR TITLE
fixed stress visualization for elastic materials

### PR DESCRIPTION
**Pull request by bobmyhill**
This PR (hopefully) fixes the stress visualization for materials where elasticity is enabled. The ve_stress_xx compositional fields correspond to the full deviatoric stress, not only additional elastic stresses.

Opened for testing.

* [x] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).
* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/master/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/master/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

**Comment by anne-glerum** [comment]
Hi @bobmyhill , thanks for the fix! Two comments:
* no test broke, is there no test with elasticity and the stress visu output? Perhaps you can add the visu output to an existing test?
* there are currently 2 other visu postprocessors that compute the stress and take elasticity into account:
 - principle_stresses.cc
 - shear_stress.cc (which I've been meaning to fix, I think it's the deviatoric stress and the elastic stresses should not be added as a last step). 
 Could you make the same changes there?

**Review by anne-glerum — APPROVED** [review]

**Comment by bobmyhill** [comment]
Hi @anne-glerum, sure, I can make those changes. 
Can we use one of your tests (i.e. one where we know what the output should look like, and which is broken in the existing code)?

**Comment by anne-glerum** [comment]
Great. If #4369 is merged, you could add the stress postprocessor to the viscoelastic_stress_build-up test (or stick with the stress invariant postprocessor already selected). The value of ve_stress_xx should be 100 MPa, ve_stress_yy -100 MPa and of the second invariant 100 MPa.

**Comment by naliboff** [comment]
@bobmyhill - I agree with @anne-glerum that the proposed changes look good and we should merge after the requested changes to principal_stresses.cc and shear_stress.cc are are done (either here or in a separate PR). If it would help, I can make separate PRs for those other post processors, which can then be merged after this PR is merged.

**Review by gassmoeller — APPROVED** [review]
Ouch, it looks like this bugfix was ready all the time and we forgot to merge it. Let's do this now.

@bobmyhill or @anne-glerum: Could you check the status of the other postprocessors mentioned in your discussion. Are they already fixed, or also still broken? If they also have waiting PRs, could you point me to them?

**Review by tjhei — APPROVED** [review]

**Comment by bobmyhill** [comment]
@gassmoeller I've now added the shear stress postprocessor and a test.
I'm confused by the test results - https://raw.githubusercontent.com/geodynamics/aspect/3391ba82148cb77f8cbf05c072321f5564235cb3/tests/viscoelastic_stress_build-up/solution/solution-00000.0000.gnuplot
The stress_xx, stress_yy and stress_xy fields look as expected, but the shear stress fields look like they either have the wrong sign, or are in the wrong order. I would have expected them to have the order xx, xy, yx, yy.

**Comment by bobmyhill** [comment]
Ah, I misunderstood the output. stress_xx, stress_xy and stress_yy are output directly from the fields, so they have the opposite sign to the ones calculated using the postprocessors.

This looks a bit weird, but I guess we have to put up with it.

**Comment by bobmyhill** [comment]
One thing that I still need to double check and document is the sign convention for the initial conditions for the stress fields.

**Comment by bobmyhill** [comment]
@anne-glerum, @gassmoeller, @tjhei: This should now be ready for a ?final round of review.

Philosophical question: at the moment, users input initial stresses with the engineering sign convention (See, for example, [aspect/tests/viscoelastic_stress_build-up.prm](https://github.com/geodynamics/aspect/blob/main/tests/viscoelastic_stress_build-up.prm)), but we output with the geological sign convention. Not something to deal with in this PR, but I do think we should make the user-facing bits of ASPECT consistent. The easiest way to do this would be to output stresses with the engineering convention.

**Comment by bobmyhill** [comment]
Hey @gassmoeller, could you look this over now that your PR (#5480) is merged?

File: source/postprocess/visualization/shear_stress.cc
Comment (gassmoeller) [inline source/postprocess/visualization/shear_stress.cc:90]: The `-=` assumes the value is 0 before this operation. However, we do not explicitly initialize it to 0 and I keep forgetting which C++ variables are default initialized to 0 and which ones arent. Let's not rely on the fact that SymmetricTensors are default initialized to 0, and instead set the value as in `shear_stress[x][y] = - in.composition ...`.

File: source/postprocess/visualization/shear_stress.cc
Comment (gassmoeller) [inline source/postprocess/visualization/shear_stress.cc:104]: To be consistent with the spaces above and also my comment above:

```suggestion
                shear_stress = -2. * eta * deviatoric_strain_rate;
```

File: source/postprocess/visualization/stress.cc
Comment (gassmoeller) [inline source/postprocess/visualization/stress.cc:104]:
```suggestion
                stress -= 2. * eta * deviatoric_strain_rate;
```

File: source/postprocess/visualization/stress.cc
Comment (gassmoeller) [inline source/postprocess/visualization/stress.cc:75]: Here you initialize the stress, so we can keep the operations below as you have them now.

**Review by gassmoeller — COMMENTED** [review]
Thanks for the reminder. Almost good to go if you address my nitpicking :-)

**Comment by bobmyhill** [comment]
Nitpicking addressed :)

**Review by gassmoeller — APPROVED** [review]

**Comment by bobmyhill** [comment]
Looks like the test failures are unrelated to this PR (gmg related). See prebuild tests here: https://github.com/geodynamics/aspect/actions/runs/7038002009/job/19156376251?pr=4375#logs

**Comment by gassmoeller** [comment]
Yup, the test failures are unrelated. This is good to go. Thanks for the patience.

**Pull request merged**
